Assignment4/JAVSIE-y160338/task08.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+"""Task08.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1oKew4DQY94e6Vp18KHGKBhHrgcOK65sd
+
+**Task 08: Completing missing data**
+"""
+
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+from rdflib import Graph, Namespace, Literal, URIRef
+g1 = Graph()
+g2 = Graph()
+g1.parse(github_storage+"/rdf/data01.rdf", format="xml")
+g2.parse(github_storage+"/rdf/data02.rdf", format="xml")
+
+from rdflib.namespace import RDF, RDFS
+from rdflib.plugins.sparql import prepareQuery
+
+VCARD = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+ns= Namespace("http://data.org#")
+print("################################################ DATA WE HAVE ################################################\nG1 :")
+for (s,p,o) in g1.triples((None,None,None)):
+  print(s,p,o)
+print("\n")
+print("\n")
+for (s,p,o) in g1.triples((None,RDF.type,ns.Person)):
+  print("G1 Person:",s,p,o)
+  print("\nG2 Data:")
+  for (s2,p2,o2) in g2.triples((s,None,None)):
+    print(s2,p2,o2)
+  print("\n")
+  print("\n")
+print("##############################################################################################################\n")
+
+for (s,p,o) in g1.triples((None,RDF.type,ns.Person)):
+  q1 = prepareQuery('''
+  SELECT ?Given ?EMAIL ?Family WHERE { 
+   ?s vcard:Given ?Given. 
+   ?s vcard:EMAIL ?EMAIL. 
+   ?s vcard:Family ?Family
```

Comment:
This is still not done correctly. You need to use optionals in the query, otherwise the results may not be complete. This query is requesting entried that have given, email and family values. If someone does not have an email, the result won't be returned